sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java

```diff
@@ -693,6 +695,33 @@ void test_python_version_parameter_no_warning() {
     verify(analysisWarning, times(0)).addUnique(PythonSensor.UNSET_VERSION_WARNING);
   }
 
+  void setup_typing_concise_rule(String pythonVersion){
+    context.fileSystem().add(inputFile("python-version/typing.py"));
+
+    activeRules = new ActiveRulesBuilder().addRule(new NewActiveRule.Builder()
+      .setRuleKey(RuleKey.of(CheckList.REPOSITORY_KEY, "S6546"))
+      .build()).build();
+
+    context.setSettings(new MapSettings().setProperty("sonar.python.version", pythonVersion));
+    sensor().execute(context);
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"2.7", "3", "3.1", "3.9"})
+  void test_python_version_pre_310(String version) {
+    setup_typing_concise_rule(version);
+
+    assertThat(context.allIssues()).isEmpty();
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"3.10", "3.11", "3.12"})
+  void test_python_version_post_310(String version) {
+    setup_typing_concise_rule(version);
+
+    assertThat(context.allIssues()).hasSize(1);
+  }
```

Comment:
Looks good! Maybe we could add a test when the version is not in our list for example `3.111` we should default to `3` I think.